src/core/qfieldcloudprojectsmodel.cpp

```diff
@@ -359,246 +368,492 @@ void QFieldCloudProjectsModel::cancelDownloadProject( const QString &projectId )
     project->downloadFileTransfers.remove( fileName );
   }
 
-  if ( project->apiNetworkReply )
-  {
-    project->apiNetworkReply->abort();
-  }
+  QgsMessageLog::logMessage( QStringLiteral( "Download of project id \"%1\" aborted" ).arg( projectId ) );
 
+  project->errorStatus = NoErrorStatus;
+  project->isPackagingActive = false;
+  project->isPackagingFailed = true;
+  project->isPackagingAborted = true;
+  project->packagingStatusString = tr( "aborted" );
   project->status = ProjectStatus::Idle;
 
-  emit dataChanged( projectIndex, projectIndex, QVector<int>() << StatusRole << PackagingStatusRole );
+  emit dataChanged( projectIndex, projectIndex, QVector<int>() << StatusRole << ErrorStatusRole << PackagingErrorStatus );
 }
 
-void QFieldCloudProjectsModel::downloadProject( const QString &projectId, bool overwriteProject )
+void QFieldCloudProjectsModel::projectRefreshData( const QString &projectId, const ProjectRefreshReason &refreshReason )
 {
   if ( !mCloudConnection )
     return;
 
   const QModelIndex projectIndex = findProjectIndex( projectId );
   if ( !projectIndex.isValid() )
     return;
+
   CloudProject *project = mProjects[projectIndex.row()];
 
-  // before downloading, the project should be idle
-  if ( project->status != ProjectStatus::Idle )
-    return;
+  NetworkReply *reply = mCloudConnection->get( QStringLiteral( "/api/v1/projects/%1/" ).arg( projectId ) );
+  connect( reply, &NetworkReply::finished, reply, [=]() {
+    if ( !findProject( projectId ) )
+      return;
 
-  // before downloading, there should be no local modification, otherwise it will be overwritten
-  if ( project->modification & LocalModification && !overwriteProject )
-    return;
+    QNetworkReply *rawReply = reply->reply();
 
-  project->packagingStatus = PackagingUnstartedStatus;
-  project->packagingStatusString = QString();
-  project->packagedLayerErrors.clear();
-  project->downloadFileTransfers.clear();
-  project->downloadFilesFinished = 0;
-  project->downloadFilesFailed = 0;
-  project->downloadBytesTotal = 0;
-  project->downloadBytesReceived = 0;
-  project->downloadProgress = 0;
+    reply->deleteLater();
```

Comment:
Should we set `reply` to `nullptr` here or ?

Reply:
I use it the very next line. If I move it,  it can work too, but does it really make a difference.